### PR TITLE
fix: Date picker timezone handling

### PR DESCRIPTION
**Pull request by MsOctoPunk**
## Summary
Fixes date picker showing incorrect dates when user timezone differs from server timezone.

## Changes
- Parse dates as UTC instead of local time
- Add timezone indicator to date picker
- Store dates in ISO 8601 format consistently
- Add utility functions for date conversion

## Test Plan
- [x] Unit tests for date conversion utilities
- [x] Manual testing across multiple timezones
- [x] Tested edge cases (DST transitions, UTC+14, etc.)

## Screenshots
_If applicable, add screenshots_

## Related Issues
Closes #

## Checklist
- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Tests added/updated
- [ ] Documentation updated

### CODE REVIEW

**Comment by MsOctoPunk** [comment]
@DrOctoPunk Could you take a look at this when you have a moment?

File: src/lib/utils/dateUtils.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/**
+ * Date utilities for consistent timezone handling
+ */
+
+/**
```
Comment (DrOctoPunk) [inline src/lib/utils/dateUtils.ts:5]: Nice!

**Review by DrOctoPunk — APPROVED** [review]
Nice!
